connection.go

```diff
@@ -799,12 +894,37 @@ func (conn *Connection) reader(r *bufio.Reader, c net.Conn) {
 				conn.markDone(fut)
 			}
 		}
+
 		if fut == nil {
 			conn.opts.Logger.Report(LogUnexpectedResultId, conn, resp)
 		}
 	}
 }
 
+// eventer goroutine gets watch events and updates values for watchers.
+func (conn *Connection) eventer(events <-chan connWatchEvent) {
+	for {
+		event, ok := <-events
+		if !ok {
+			// The channel is closed.
+			break
+		}
+
+		if value, ok := conn.watchMap.Load(event.key); ok {
```

Comment:
If I got it right and we do not expect unknown keys, I think it shouldn't hurt to log it.

Reply:
I think there is a possible situation here (due to network, OS or runtime):

1. A connector sends IPROTO_WATCH.
2. A Tarantool instance sends IPROTO_EVENT.
3. The connector sends IPROTO_UNWATCH.
4. The connector receives IPROTO_EVENT.

A log message will be confusing in this expected case.

Reply:
I'm not a big fan of ignoring possible error case, but... It doesn't seems really useful to log unknown keys even if Tarantool starts to send spontaneous unexpected messages fr any reason. So maybe we really should skip them like we skip unknown keys in responses.

Reply:
I've been executing tests in practice:

```
    1. A connector sends IPROTO_WATCH.
    2. A Tarantool instance sends IPROTO_EVENT.
    3. The connector sends IPROTO_UNWATCH.
    4. The connector receives IPROTO_EVENT.
```

The situation is quite easy to reproduce in practice, we simply spam the user with log messages that are not errors actually. So I've added a comment in the code instead of a log message.